test/v3/controllers/requestParsers/validators/RetrieveSelfEmploymentValidatorSpec.scala

```diff
@@ -32,26 +32,57 @@ class RetrieveSelfEmploymentValidatorSpec extends UnitSpec {
   "validator" should {
     "return no errors" when {
       "passed valid raw data" in {
-        val input = RetrieveSelfEmploymentBsasRawData(validNino, validCalculationId)
+        val input = RetrieveSelfEmploymentBsasRawData(validNino, validCalculationId, None)
+        validator.validate(input) shouldBe List()
+      }
+      "a valid request with a taxYear is supplied" in {
+        val input = RetrieveSelfEmploymentBsasRawData(validNino, validCalculationId, Some("2023-24"))
         validator.validate(input) shouldBe List()
       }
     }
     "return a single error" when {
       "passed raw data with an invalid nino" in {
-        val input = RetrieveSelfEmploymentBsasRawData(invalidNino, validCalculationId)
+        val input = RetrieveSelfEmploymentBsasRawData(invalidNino, validCalculationId, None)
         validator.validate(input) shouldBe List(NinoFormatError)
       }
       "passed raw data with an invalid calculation id" in {
-        val input = RetrieveSelfEmploymentBsasRawData(validNino, invalidCalculationId)
+        val input = RetrieveSelfEmploymentBsasRawData(validNino, invalidCalculationId, None)
         validator.validate(input) shouldBe List(CalculationIdFormatError)
       }
     }
     "return multiple errors" when {
       "passed raw data with multiple invalid fields" in {
-        val input = RetrieveSelfEmploymentBsasRawData(invalidNino, invalidCalculationId)
+        val input = RetrieveSelfEmploymentBsasRawData(invalidNino, invalidCalculationId, None)
         validator.validate(input) shouldBe List(NinoFormatError, CalculationIdFormatError)
       }
     }
+    "return InvalidTaxYearParameterError error" when {
+      "a tax year before TYS is suppled" in {
+        val input = RetrieveSelfEmploymentBsasRawData(validNino, validCalculationId, Some("2022-23"))
+
+        validator.validate(input) shouldBe
+          List(InvalidTaxYearParameterError)
+      }
+    }
+
+    "return TaxYearFormatError error" when {
+      "a badly formatted tax year is suppled" in {
+        val input = RetrieveSelfEmploymentBsasRawData(validNino, validCalculationId, Some("BAD_TAX_YEAR"))
+
+        validator.validate(input) shouldBe
+          List(TaxYearFormatError)
+      }
+    }
+
+    "return RuleTaxYearRangeInvalidError error" when {
+      "a tax year range of more than one year is supplied" in {
+        val input = RetrieveSelfEmploymentBsasRawData(validNino, validCalculationId, Some("2022-24"))
+
+        validator.validate(input) shouldBe
+          List(RuleTaxYearRangeInvalidError)
+      }
+    }
+
```

Comment:
I know the code was not here before, but worth maybe testing the other potential errors too (`NinoFormatError` & `CalculationIdFormatError`)

Reply:
Those are already tested in "return a single error" scenario